linera-chain/src/manager.rs

```diff
@@ -399,18 +399,15 @@ impl ChainManager {
     pub fn create_vote(
         &mut self,
         proposal: BlockProposal,
-        outcome: BlockExecutionOutcome,
+        executed_block: ExecutedBlock,
         key_pair: Option<&KeyPair>,
         local_time: Timestamp,
+        blobs: BTreeMap<BlobId, Blob>,
     ) -> Option<Either<&Vote<ValidatedBlock>, &Vote<ConfirmedBlock>>> {
-        // Record the proposed block, so it can be supplied to clients that request it.
-        self.proposed = Some(proposal.clone());
-        self.update_current_round(local_time);
-        let ProposalContent { block, round, .. } = proposal.content;
-        let executed_block = outcome.with(block);
+        let round = proposal.content.round;
 
         // If the validated block certificate is more recent, update our locked block.
-        if let Some(lite_cert) = proposal.validated_block_certificate {
+        if let Some(lite_cert) = proposal.validated_block_certificate.clone() {
```

Comment:
I'd remove the `clone()` here and move it to line 417 - so that it's cloned when it's really necessary.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3006/commits/6ac957b6919b273df5ebe6c15ce2e069645b6bf5.